specification/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/stable/2023-09-04/redhatopenshift.json

```diff
@@ -0,0 +1,2566 @@
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/OpenShiftVersionProperties",
+          "description": "The properties for the OpenShiftVersion resource.",
+          "x-ms-client-flatten": true
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/systemData",
+          "description": "The system meta data relating to this resource.",
+          "readOnly": true
+        }
+      }
+    },
+    "OpenShiftVersionList": {
+      "description": "OpenShiftVersionList represents a List of available versions.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The List of available versions.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/OpenShiftVersion"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "description": "Next Link to next operation.",
+          "type": "string"
+        }
+      }
+    },
+    "OpenShiftVersionProperties": {
+      "description": "OpenShiftVersionProperties represents the properties of an OpenShiftVersion.",
+      "type": "object",
+      "properties": {
+        "version": {
+          "description": "Version represents the version to create the cluster at.",
+          "type": "string"
+        }
+      }
+    },
+    "Operation": {
+      "description": "Operation represents an RP operation.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Operation name: {provider}/{resource}/{operation}.",
+          "type": "string"
+        },
+        "display": {
+          "$ref": "#/definitions/Display",
+          "description": "The object that describes the operation."
+        },
+        "origin": {
+          "description": "Sources of requests to this operation.  Comma separated list with valid values user or system, e.g. \"user,system\".",
+          "type": "string"
+        }
+      }
+    },
+    "OperationList": {
+      "description": "OperationList represents an RP operation list.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "List of operations supported by the resource provider.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Operation"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "description": "The link used to get the next page of operations.",
+          "type": "string"
+        }
+      }
+    },
+    "OutboundType": {
+      "description": "The outbound routing strategy used to provide your cluster egress to the internet.",
+      "enum": [
+        "Loadbalancer",
+        "UserDefinedRouting"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "OutboundType",
+        "modelAsString": true
+      }
+    },
+    "PreconfiguredNSG": {
```

Comment:
>PreconfiguredNSG [](http://example.com/codeflow?start=5&length=16)

It should be something like "PreconfiguredNSGState" for more accurate representation of its meaning.

Reply:
In order to maintain the consistency with other fields naming convention (eg: EncryptionAtHost, FipsValidatedModules) that hold the similar enum values it was decided to name it this way during design.